File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/util/Preconditions.java
package io.casey.musikcube.remote.util;

import android.os.Looper;

public class Preconditions {
    public static void throwIfNotOnMainThread() {
        if (Thread.currentThread() != Looper.getMainLooper().getThread()) {
            throw new IllegalStateException("not on main thread");
        }
    }
}

File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/util/Preconditions.kt
package io.casey.musikcube.remote.util

import android.os.Looper

object Preconditions {
    fun throwIfNotOnMainThread() {
        if (Thread.currentThread() !== Looper.getMainLooper().thread) {
            throw IllegalStateException("not on main thread")
        }
    }
}
